src/bimorph_mirror_analysis/__main__.py

```diff
@@ -13,37 +12,25 @@
 
 __all__ = ["main"]
 
+app = typer.Typer()
 
-def main(args: Sequence[str] | None = None) -> None:
-    """Argument parser for the CLI."""
-    parser = ArgumentParser()
-    parser.add_argument(
-        "-v",
-        "--version",
-        action="version",
-        version=__version__,
-    )
-    parser.add_argument(
-        "file_path",
-        type=str,
-        help="Path to the file containing the output of the Bluesky plan.",
-    )
-    parser.add_argument(
-        "-o",
-        "--output",
-        type=str,
-        help="Path to save the optimal voltages.",
-    )
-    a = parser.parse_args(args)
-    file_path = a.file_path
-    file_type = a.file_path.split(".")[-1]
+
+@app.command(name=None)
+def calculate_voltages(
+    file_path: str = typer.Argument(help="The path to the csv file to be read."),
+    output_path: str | None = typer.Option(
+        None,
+        help="The path to save the output\
+optimal voltages to, optional.",
+    ),
+):
+    file_type = file_path.split(".")[-1]
     optimal_voltages = calculate_optimal_voltages(file_path)
     optimal_voltages = np.round(optimal_voltages, 2)
     date = datetime.datetime.now().date()
-    if a.output:
-        output_path = a.output
-    else:
-        output_path = f"{a.file_path.replace(f'.{file_type}', '')}\
+
+    if output_path is None:
```

Comment:
Should: add test to exercise both paths of this (currently I think you only test the path where these is no `output_path`)